crawl/pom.py

```diff
@@ -297,17 +292,33 @@ def __init__(self, workspace, artifact_def, dependency):
         super(TemplatePomGen, self).__init__(workspace, artifact_def, dependency)
     def gen(self, pomcontenttype):
         pom_content = self.artifact_def.custom_pom_template_content
-        pom_content, parsed_dependencies = self._process_pom_template_content(pom_content)
-
-        properties = self._get_properties(pomcontenttype, parsed_dependencies)
+        pom_content, parsed_dependencies, parsed_properties = self._process_pom_template_content(pom_content)
+        all_version_properties, template_version_properties, group_version_dict = self._get_version_properties(pomcontenttype, parsed_properties)
+        for k in all_version_properties.keys():
+            pom_content = pom_content.replace("#{%s}" % k, all_version_properties[k])
+        generate_properties_section = "#{pomgen.generate_properties}"
+        properties_section_start_index = pom_content.find(TemplatePomGen.PROPERTIES_SECTION_START)
+        substitute_version_properties = False
+        if generate_properties_section in pom_content or properties_section_start_index != -1:
+            substitute_version_properties = True
+        initial_properties, updated_group_version_dict = self._get_crawled_dependencies_properties(pomcontenttype, parsed_dependencies, group_version_dict, substitute_version_properties)
+        version_properties_content = pomproperties.gen_version_properties(updated_group_version_dict, pom_content)
+
+        if generate_properties_section in pom_content:
+            content = ""
+            content, indent = common.xml(content, "properties", indent=_INDENT)
+            content += version_properties_content
+            content, indent = common.xml(content, "properties", indent, close_element=True)
+            pom_content = pom_content.replace(generate_properties_section, content)
+        elif properties_section_start_index != -1:
+            properties_section_end_index = pom_content.index(TemplatePomGen.PROPERTIES_SECTION_END)
+            inject_index = pom_content[:properties_section_end_index].rfind(os.linesep) + 1
+            pom_content = pom_content[:inject_index] + version_properties_content + pom_content[inject_index:]
```

Comment:
Can this logic that generates the properties content live in a different method for better readability?